Framework/Core/include/Framework/GroupedCombinations.h

```diff
@@ -0,0 +1,296 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef FRAMEWORK_GROUPEDCOMBINATIONS_H
+#define FRAMEWORK_GROUPEDCOMBINATIONS_H
+
+#include "Framework/ASoAHelpers.h"
+#include "Framework/GroupSlicer.h"
+#include "Framework/StringHelpers.h"
+
+namespace o2::framework
+{
+
+// Define the type of a tuple interleaved from 2 tuples
+// Base struct without specialization
+template <typename T1, typename T2>
+struct InterleavedTupleType {
+  template <typename... Is>
+  struct TupleType {
+    using type = std::false_type;
+  };
+  template <std::size_t... Is>
+  struct TupleType<std::index_sequence<Is...>> {
+    using type = std::tuple<pack_element_t<Is, interleaved_pack_t<pack<T1>, pack<T2>>>...>;
+  };
+  using type = typename TupleType<std::make_index_sequence<2>>::type;
+};
+
+// Proper specialization with tuples
+template <typename... T1s, typename... T2s>
+struct InterleavedTupleType<std::tuple<T1s...>, std::tuple<T2s...>> {
+  template <typename... Is>
+  struct TupleType {
+    using type = std::false_type;
+  };
+  template <std::size_t... Is>
+  struct TupleType<std::index_sequence<Is...>> {
+    using type = std::tuple<pack_element_t<Is, interleaved_pack_t<pack<T1s...>, pack<T2s...>>>...>;
+  };
+  using type = typename TupleType<std::make_index_sequence<sizeof...(T1s) + sizeof...(T2s)>>::type;
+};
+
+// Create an instance of a tuple interleaved from given tuples
+template <typename... T1s, typename... T2s, std::size_t... Is>
+auto interleaveTuplesImpl(std::tuple<T1s...>& t1, std::tuple<T2s...>& t2, std::index_sequence<Is...>)
```

Comment:
Why not simply introducing a `pack_to_tuple(pack<...>)` helper, with the associated `pack_to_tuple_t<pack<...>>`? Then expressing the interleaved tuple should be just a matter of using those helpers, no? I think this part would become greatly simplified. Could you give it a try?